swarm/storage/mock/mem/mem.go

```diff
@@ -75,11 +87,33 @@ func (s *GlobalStore) Put(addr common.Address, key []byte, data []byte) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	if _, ok := s.nodes[string(key)]; !ok {
-		s.nodes[string(key)] = make(map[common.Address]struct{})
+	if i, found := s.nodeKeyIndex(addr, key); !found {
```

Comment:
I find, again, these names as confusing, e.g .`nodeKeyIndex` vs `keyNodeIndex`

Reply:
Could you provide naming that would be more clear to you? That would help me a lot, instead of guessing. I tried to keep them as short as possible with comments on them explaining what they do.